Samples/DirectMLNpuInference/main.cpp

```diff
@@ -13,11 +13,46 @@
 
 using Microsoft::WRL::ComPtr;
 
-void InitializeDirectML(ID3D12Device1** d3dDeviceOut, ID3D12CommandQueue** commandQueueOut, IDMLDevice** dmlDeviceOut) {
-    // Whether to skip adapters which support Graphics in order to target NPU for testing
-    bool forceComputeOnlyDevice = true;
-    bool forceGenericMLDevice = false;
-    
+bool TryGetProperty(ComPtr<IDXCoreAdapter>& adapter, DXCoreAdapterProperty prop, std::string& outputValue)
```

Comment:
Input pointer parameters should just be passed directly...

```suggestion
bool TryGetProperty(IDXCoreAdapter* adapter, DXCoreAdapterProperty prop, std::string& outputValue)
```

- (perf) it's an unnecessary indirection of pointers to pointers.
- (privilege principle) We shouldn't pass the containing ComPtr's reference unless the function itself actually needs to know that because it modifies the ComPtr in some way (otherwise it has no privilege to know such things about the caller implementation).
- ([robustness principle](https://en.wikipedia.org/wiki/Robustness_principle)) Functions (especially ones in samples, which tend to be copied around by people) should accept the more general thing (`Foo*`) rather than expecting specifically a `Microsoft::WRL::ComPtr<Foo>`, as the caller may use a different ComPtr or pass raw pointers.

Reply:
Ah I was wondering if it was right to pass the ComPtr like that. Thanks for explaining the reasoning! Fixed here and in another area of the code.